src/nvim/mbyte.c

```diff
@@ -1859,7 +1859,7 @@ int mb_tail_off(const char_u *base, const char_u *p)
     }
   }
 
-  if (utf8len_tab[p[-j]] != i + j + 1) {
+  if (utf8len_tab[(uint8_t)p[-j]] != i + j + 1) {
```

Comment:
are the usages of `p[...]` above also safe? we could follow the `p_in` pattern to not need to worry.

Reply:
I think so. Sure.